testnet/stacks-node/src/config.rs

```diff
@@ -363,6 +370,163 @@ pub struct Config {
+            .map(|m| m.modified())
+            .and_then(convert::identity)
+        {
+            Ok(t) => t,
+            Err(e) => {
+                warn!("Failed to get modified time of config file: {}", e);
+                return;
+            }
+        };
+
+        self.load_time = config_file_time;
+        if config_file_time <= self.load_time {
+            return;
+        }
+
+        let config = Self::load(&config_path);
+        match config {
+            Ok(new_config) => {
+                info!("Loaded config at {}", config_path);
+                let config_handle = &self.config_handle;
+                let old_config = config_handle.get();
+                Self::diff_config(&old_config, &new_config);
+                config_handle.replace(&new_config);
+            }
+            Err(e) => warn!("Failed to reload config: {}", e),
+        }
+    }
+
+    fn load(config_path: &String) -> Result<Config, String> {
+        let config_file = ConfigFile::from_path(&config_path)
+            .map_err(|e| format!("Failed to load config file: {}", e))?;
+        let config = Config::from_config_file_with_options(config_file.clone(), false)
+            .map_err(|e| format!("Failed to parse config file: {}", e))?;
+        Ok(config)
+    }
+
+    fn diff_config(old_config: &Config, new_config: &Config) {
+        Self::visit_diff(
+            &old_config,
+            &new_config,
+            |s| &s.burnchain.burn_fee_cap,
+            "burnchain.burn_fee_cap",
+        );
+        Self::visit_diff(
+            &old_config,
+            &new_config,
+            |s| &s.burnchain.satoshis_per_byte,
+            "burnchain.satoshis_per_byte",
+        );
+        Self::visit_diff(
+            &old_config,
+            &new_config,
+            |s| &s.burnchain.rbf_fee_increment,
+            "burnchain.rbf_fee_increment",
+        );
+        Self::visit_diff(
+            &old_config,
+            &new_config,
+            |s| &s.burnchain.max_rbf,
+            "burnchain.max_rbf",
+        );
+        Self::visit_diff(
+            &old_config,
+            &new_config,
+            |s| &s.node.microblock_frequency,
+            "node.microblock_frequency",
+        );
+        Self::visit_diff(
+            &old_config,
+            &new_config,
+            |s| &s.node.wait_time_for_microblocks,
+            "node.wait_time_for_microblocks",
+        );
+    }
+
+    fn visit_diff<Struct, Field: PartialEq + std::fmt::Display>(
+        struct1: &Struct,
+        struct2: &Struct,
+        extractor: fn(&Struct) -> &Field,
+        field_name: &str,
+    ) {
+        let field1 = extractor(struct1);
+        let field2 = extractor(struct2);
+        if field1 != field2 {
+            info!("{} changed from {} to {}", field_name, field1, field2)
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct ConfigLoaderHandle {
+    handle: std::sync::Arc<std::sync::Mutex<RefCell<ConfigLoader>>>,
```

Comment:
You probably don't need `RefCell` here.

Reply:
RefCell removed from `ConfigHandle` and `ConfigLoaderHandle`.